eventsources/sources/pulsar/start.go

```diff
@@ -0,0 +1,146 @@
+	return el.EventSourceName
+}
+
+// GetEventName returns name of event
+func (el *EventListener) GetEventName() string {
+	return el.EventName
+}
+
+// GetEventSourceType return type of event server
+func (el *EventListener) GetEventSourceType() apicommon.EventSourceType {
+	return apicommon.PulsarEvent
+}
+
+// StartListening listens NSQ events
+func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byte) error) error {
+	log := logging.FromContext(ctx).
+		With(logging.LabelEventSourceType, el.GetEventSourceType(), logging.LabelEventName, el.GetEventName())
+	log.Info("started processing the Pulsar event source...")
+	defer sources.Recover(el.GetEventName())
+
+	msgChannel := make(chan pulsar.ConsumerMessage)
+
+	pulsarEventSource := &el.PulsarEventSource
+
+	subscriptionType := pulsar.Exclusive
+	if pulsarEventSource.Type == "shared" {
+		subscriptionType = pulsar.Shared
+	}
+
+	log.Info("setting consumer options...")
+	consumerOpt := pulsar.ConsumerOptions{
+		Topics:           pulsarEventSource.Topics,
+		SubscriptionName: el.EventName,
+		Type:             subscriptionType,
+		MessageChannel:   msgChannel,
+	}
+
+	log.Info("setting client options...")
+	clientOpt := pulsar.ClientOptions{
+		URL:                        pulsarEventSource.URL,
+		TLSTrustCertsFilePath:      pulsarEventSource.TLSTrustCertsFilePath,
+		TLSAllowInsecureConnection: pulsarEventSource.TLSAllowInsecureConnection,
+		TLSValidateHostname:        pulsarEventSource.TLSValidateHostname,
+	}
+
+	if pulsarEventSource.TLS != nil {
+		log.Info("setting tls auth option...")
+		clientOpt.Authentication = pulsar.NewAuthenticationTLS(pulsarEventSource.TLS.ClientCertPath, pulsarEventSource.TLS.ClientKeyPath)
+	}
+
+	var client pulsar.Client
+
+	if err := sources.Connect(common.GetConnectionBackoff(pulsarEventSource.ConnectionBackoff), func() error {
+		var err error
+		if client, err = pulsar.NewClient(clientOpt); err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return errors.Wrapf(err, "failed to connect to %s for event source %s", pulsarEventSource.URL, el.GetEventName())
+	}
+
+	log.Info("subscribing to messages on the topic...")
+	consumer, err := client.Subscribe(consumerOpt)
+	if err != nil {
+		return errors.Wrapf(err, "failed to connect to topic %+v for event source %s", pulsarEventSource.Topics, el.GetEventName())
+	}
+
+consumeMessages:
+	for {
+		select {
+		case msg := <-msgChannel:
+			log.Infof("received a message on the topic %s", msg.Topic())
+			payload := msg.Payload()
+			eventData := &events.PulsarEventData{
+				Key:         msg.Key(),
+				PublishTime: msg.PublishTime().UTC().String(),
+				Body:        payload,
+			}
+			if pulsarEventSource.JSONBody {
+				eventData.Body = (*json.RawMessage)(&payload)
+			}
+
+			eventBody, err := json.Marshal(eventData)
+			if err != nil {
+				log.Error("failed to marshal the event data. rejecting the event...", zap.Error(err))
```

Comment:
logging format is not right, it will combine the msg and error to one string. Use either one of following.

`log.Desugar().Error("xxxx", zap.Error(err))`

or 

`log.Errorw("xxxx", "error", err)`